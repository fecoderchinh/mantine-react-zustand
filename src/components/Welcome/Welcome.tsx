import { Title, Text, Anchor } from '@mantine/core'
import classes from './Welcome.module.css'
import { useAuthStore } from '@/store/auth'
import { useNavigate } from 'react-router'
import { notifications } from '@mantine/notifications'

export const Welcome = () => {
   const authState = useAuthStore()

   const navigate = useNavigate()

   const handleLogout = async () => {
      await authState.logout().then(res => {
         if (!res.isAuthenticated) {
            notifications.show({
               title: 'Sign out success!',
               message: 'We will miss you!',
               color: 'green',
            })
            navigate('/')
         }
      })
   }
   return (
      <>
         <Title className={classes.title} ta='center' mt={100}>
            Welcome to{' '}
            <Text inherit variant='gradient' component='span' gradient={{ from: 'pink', to: 'yellow' }}>
               Mantine
            </Text>
         </Title>
         <Text color='dimmed' ta='center' size='lg' maw={580} mx='auto' mt='xl'>
            This starter Vite project includes a minimal setup, if you want to learn more on Mantine + Vite integration
            follow{' '}
            <Anchor href='https://mantine.dev/guides/vite/' size='lg'>
               this guide
            </Anchor>
            . To get started edit pages/Home.page.tsx file.
         </Text>
         <button onClick={handleLogout}>Log out</button>
      </>
   )
}

export default Welcome
