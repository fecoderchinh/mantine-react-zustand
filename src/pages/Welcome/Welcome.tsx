import { lazy } from 'react'

const ColorSchemeToggle = lazy(() => import('@/components/ColorSchemeToggle/ColorSchemeToggle'))
const Welcome = lazy(() => import('@/components/Welcome/Welcome'))

export const PageWelcome = () => {
   return (
      <>
         <Welcome />
         <ColorSchemeToggle />
      </>
   )
}

export default PageWelcome
