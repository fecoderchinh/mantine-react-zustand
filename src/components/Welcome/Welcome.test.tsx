import { render } from '@tests'
import { Welcome } from './Welcome'

describe('Welcome component', () => {
   it('has correct Vite guide link', () => {
      const { container } = render(<Welcome />)
      expect(container).toMatchSnapshot()
   })
})
