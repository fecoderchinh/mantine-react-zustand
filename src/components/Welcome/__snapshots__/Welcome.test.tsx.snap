// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component has correct Vite guide link 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <h1
    class="title m-8a5d1357 mantine-Title-root"
    data-order="1"
    style="--title-fw: var(--mantine-h1-font-weight); --title-lh: var(--mantine-h1-line-height); --title-fz: var(--mantine-h1-font-size); text-align: center;"
  >
    Welcome to
     
    <span
      class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
      data-inherit="true"
      data-variant="gradient"
      style="--text-gradient: linear-gradient(45deg, var(--mantine-color-pink-filled) 0%, var(--mantine-color-yellow-filled) 100%);"
    >
      Mantine
    </span>
  </h1>
  <p
    class="mantine-focus-auto m-b6d8b162 mantine-Text-root"
    color="dimmed"
    data-size="lg"
    style="--text-fz: var(--mantine-font-size-lg); --text-lh: var(--mantine-line-height-lg); --text-color: dimmed; margin-right: auto; margin-left: auto; text-align: center; max-width: calc(36.25rem * var(--mantine-scale));"
  >
    This starter Vite project includes a minimal setup, if you want to learn more on Mantine + Vite integration follow
     
    <a
      class="mantine-focus-auto m-849cf0da m-b6d8b162 mantine-Text-root mantine-Anchor-root"
      data-size="lg"
      data-underline="hover"
      href="https://mantine.dev/guides/vite/"
      style="--text-fz: var(--mantine-font-size-lg); --text-lh: var(--mantine-line-height-lg);"
    >
      this guide
    </a>
    . To get started edit pages/Home.page.tsx file.
  </p>
</div>
`;
